### PR TITLE
Create models User, Role, Permission, and make relations between them 

### PR DESCRIPTION
**Pull request by darkgraycat**
Sequelize models:
User (uuid, name email, createdAt, updatedAt, Role)
Role (uuid, name, Permissons[])
Permission (uuid, type = enum)
RolePermission(Role, Permisson)

### CODE REVIEW

File: src/models/Permission.model.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { BelongsToMany, Column, IsUUID, Model, PrimaryKey, Table, Unique } from "sequelize-typescript";
```
Comment (CatDidenko) [inline src/models/Permission.model.ts]: Use single quotes, please

File: src/models/Permission.model.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  @Column
+  public uuid: string
+
+  @Unique
+  @Column
+  public type: string
+
+  @BelongsToMany(() => Role, () => RolePermission)
+  public roles: Role[]
+}
```
Comment (CatDidenko) [inline src/models/Permission.model.ts]: Where is semicolon after each instruction? Should be added.

File: src/models/Role.model.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+interface RoleAttributes {
+  uuid: string
+  name: string
+  users?: User[]
+}
+
+@Table
+export default class Role extends Model<RoleAttributes> {
+
```
Comment (CatDidenko) [inline src/models/Role.model.ts]: Remove blank line

File: src/models/Permission.model.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+interface PermissionAttributes {
+  uuid: string
+  type: OPERATION
+}
+
+@Table
+export default class Permission extends Model<PermissionAttributes> {
+
```
Comment (CatDidenko) [inline src/models/Permission.model.ts]: Remove blank line

File: src/models/RolePermission.model.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+@Table
+export default class RolePermission extends Model<RolePermission> {
+  @ForeignKey(() => Role)
+  @Column({ field: 'role_uuid' })
+  public roleUuid: string
+
+  @ForeignKey(() => Permission)
+  @Column({ field: 'permission_uuid' })
+  public permissionUuid: string
+}
```
Comment (CatDidenko) [inline src/models/RolePermission.model.ts]: Add blank line in the end of file

File: src/sequelize.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import { Sequelize } from 'sequelize-typescript'
+
+export default new Sequelize({
+  database: 'usersdb',
+  username: 'root',
+  password: 'msql1234',
+  host: 'localhost',
+  dialect: 'mysql',
```
Comment (CatDidenko) [inline src/sequelize.ts]: Move to the constants file these values.

File: src/index.ts [hunk 7]
```diff
@@ -1 +1,57 @@
...
+
+    admin.$set('permissions', [permissionToCreate, permissionToDelete, permissionToRead, permissionToUpdate])
+    user.$set('permissions', [permissionToRead])
+
+    admin.$set('users', [person_1])
+    user.$set('users', [person_2, person_3])
+
+    // await sequelize.query('UPDATE users SET roleId = 2 WHERE id = 1')
+    // await sequelize.query('UPDATE users SET roleId = 1 WHERE id = 2')
+    // await sequelize.query('UPDATE users SET roleId = 1 WHERE id = 3')
```
Comment (CatDidenko) [inline src/index.ts]: Remove unnecessary comments

File: src/models/Role.model.ts [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  @Unique
+  @Column
+  public name: string
+
+  @HasMany(() => User)
+  public users: User[]
+
+  @BelongsToMany(() => Permission, () => RolePermission)
+  public permissions: Permission[]
+
```
Comment (CatDidenko) [inline src/models/Role.model.ts]: Remove blank line

**Review by CatDidenko — CHANGES_REQUESTED** [review]

File: src/models/Permission.model.ts (hunk 1 above)
Comment (darkgraycat) [reply src/models/Permission.model.ts]: vscode auto import)

File: src/models/Role.model.ts (hunk 3 above)
Comment (darkgraycat) [reply src/models/Role.model.ts]: which 11 or 14?

File: src/sequelize.ts [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import { Sequelize } from 'sequelize-typescript';
+
+import { DATABASE, USERNAME, PASSWORD, DIALECT, HOST } from './constants'
```
Comment (CatDidenko) [inline src/sequelize.ts]: Semicolon?

File: src/models/Permission.model.ts [hunk 10]
```diff
@@ -0,0 +1,32 @@
+import { BelongsToMany, Column, Default, IsUUID, Model, PrimaryKey, Table, Unique } from 'sequelize-typescript';
+import { UUIDV4 } from 'sequelize';
+
+import Role from './role.model';
+
+export enum OPERATION {
+  CREATE = 'CREATE',
+  READ = 'READ',
+  UPDATE = 'UPDATE',
+  DELETE = 'DELETE'
```
Comment (CatDidenko) [inline src/models/Permission.model.ts]: Add comma

File: src/sequelize.ts [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+export default new Sequelize({
+  database: DATABASE,
+  username: USERNAME,
+  password: PASSWORD,
+  host: HOST,
+  dialect: DIALECT,
+  models: [__dirname + '/models'],
+  define: {
+    timestamps: false
+  }
```
Comment (CatDidenko) [inline src/sequelize.ts]: Add semicolons

**Review by CatDidenko — APPROVED** [review]

**Pull request merged**